arrow-docs/docs/optics/filterindex/README.md

```diff
@@ -11,33 +11,40 @@ permalink: /optics/filterindex/
 
 If the foci `A` for a structure `S` can be indexed by `I`, then a `Traversal` can be created by `FilterIndex` that is filtered by a predicate on `I`.
 
-`FilterIndex` can easily be created, given a `Traverse` instance and an indexing function.
+`FilterIndex` can easily be created, given a `Every` instance that filers a certain index.
 
 ```kotlin:ank
 import arrow.core.*
+import arrow.typeclasses.*
 import arrow.optics.*
 import arrow.optics.typeclasses.*
-import arrow.core.extensions.listk.traverse.*
 
-val filterIndexStringByIndex = FilterIndex.fromTraverse<ForListK, String>({ list ->
-    list.fix().map {
-        it toT it.length
+val filterIndexStringByIndex : FilterIndex<List<String>, Int, String> = FilterIndex { p ->
+  object : Every<List<String>, String> {
+    override fun <R> foldMap(M: Monoid<R>, s: List<String>, map: (String) -> R): R = M.run {
+      s.foldIndexed(empty()) { index, acc, a -> if (p(index)) acc.combine(map(a)) else acc }
     }
-}, ListK.traverse())
+  
+    override fun modify(s: List<String>, map: (focus: String) -> String): List<String> =
+      s.mapIndexed { index, a -> if (p(index)) map(a) else a }
+  }
+}
 ```
 
 Given a `FilterIndex` instance, we can create a `Traversal` that filters out the foci that do not match the predicate.
 
 ```kotlin:ank
-val filter: Traversal<ListKOf<String>, String> = filterIndexStringByIndex.filter { length -> length > 3 }
+val filter: Every<List<String>, String> = filterIndexStringByIndex.filter { length -> length > 3 }
```

Comment:
I think this example is a bit confusing when naming the parameter as `length`. It could be interpreted that we're filtering out those elements whose length is less or equal to 3. But we're actually removing the items whose index within the collection doesn't meet that criteria.
 
```suggestion
val filter: Every<List<String>, String> = filterIndexStringByIndex.filter { index -> index > 3 }
```